### PR TITLE
add config.json (in a better way)

### PR DESCRIPTION
**Pull request by interpolation-0**
adds the config.json option outlined by the readme

lmk if i fucked up

### CODE REVIEW

**Comment by interpolation-0** [comment]
i might be fucking stupid

**Comment by interpolation-0** [comment]
nvm im good

**Comment by interpolation-0** [comment]
or am i

**Comment by interpolation-0** [comment]
no im fucking stupid

File: main.py [hunk 1]
```diff
@@ -298,6 +299,13 @@ async def main(args):
   parser.add_argument("--bandwidth", default=1000, help="Bandwidth limit per IP, in kilobytes per second.")
   parser.add_argument("--connections", default=30, help="Connections limit per IP, in kilobytes per second.")
   parser.add_argument("--window", default=60, help="Fixed window length for rate limits, in seconds.")
+  parser.add_argument("--config", default="config.json", help="The config file location.")
   args = parser.parse_args()
+  def load_config():
```
Comment (ading2210) [inline main.py:304]: This shouldn't be a nested function.

**Review by ading2210 — CHANGES_REQUESTED** [review]

File: main.py [hunk 2]
```diff
@@ -298,6 +299,13 @@ async def main(args):
   parser.add_argument("--bandwidth", default=1000, help="Bandwidth limit per IP, in kilobytes per second.")
   parser.add_argument("--connections", default=30, help="Connections limit per IP, in kilobytes per second.")
   parser.add_argument("--window", default=60, help="Fixed window length for rate limits, in seconds.")
+  parser.add_argument("--config", default="config.json", help="The config file location.")
```
Comment (ading2210) [inline main.py:302]: Also, don't make this try to load a nonexistent file as a default. It should only a attempt to load a config file if the `--config` argument is passed in.

**Review by ading2210 — CHANGES_REQUESTED** [review]

**Comment by ading2210** [comment]
This PR is stale and has merge conflicts.

**Pull request closed without merging**